web/base_viewer.js

```diff
@@ -1476,6 +1478,35 @@ class BaseViewer {
     this._setCurrentPageNumber(pageNumber, /* resetCurrentPageView = */ true);
     this.update();
   }
+
+  /**
+   * @type {number} One of the values in {ReadingDirection}.
+   */
+  get direction() {
+    return this._direction;
+  }
+
+  /**
+   * @param {number} dir
+   */
+  set direction(direction) {
+    if (this._direction === direction) {
+      return;
+    }
+    if (!Object.values(ReadingDirection).includes(direction)) {
+      throw new Error(`Invalid document direction: ${direction}`);
+    }
```

Comment:
The validation needs to be done using the `isValidReadingDirection` helper function (see other review comment), and I don't *think* that we need to support setting this until the document has loaded (since that simplifies things overall):
```suggestion
  set readingDirection(direction) {
    if (!isValidReadingDirection(direction)) {
      throw new Error(`Invalid reading direction: ${direction}`);
    }
    if (!this.pdfDocument) {
      return;
    }
    if (this._readingDirection === direction) {
      return; // The reading direction didn't change.
    }
```